src/routes/relay/entities/schema/sponsored-call.schema.spec.ts

```diff
@@ -75,6 +77,34 @@ describe('sponsoredCall schema', () => {
     });
   });
 
+  it('should validate and predict the Safe address from a valid creation call', async () => {
+    const chainId = '5';
+
+    const to = getProxyFactoryDeployment({
+      network: chainId,
+    })?.defaultAddress;
+
+    const data = {
+      chainId,
+      to,
+      data: MOCK_CREATE_PROXY_WITH_NONCE_TX_CALL_DATA,
+      gasLimit: faker.random.numeric(),
+    };
+
+    const result = await SponsoredCallSchema.safeParseAsync(data);
+
+    expect(result.success).toBe(true);
+
+    if (!result.success) {
+      return;
+    }
+
+    expect(result.data).toStrictEqual({
+      ...data,
+      safeAddress: '0x8E1a7c91EF7CfE9356f2c654286e00E9ECBf5C3B',
```

Comment:
Where is this address from?

Reply:
This is the predicted address from `MOCK_CREATE_PROXY_WITH_NONCE_TX_CALL_DATA`.